lib/utils/updateCompiler.js

```diff
@@ -18,8 +18,8 @@ function updateCompiler(compiler, options) {
     } else {
       devServerPlugin = new DevServerPlugin(options);
       compilerOptions.plugins.push(devServerPlugin);
```

Comment:
No need to adding this, just use `devServerPlugin.apply(compiler);`

Reply:
I forgot the purpose of #2839, which was to support reusing a single compiler instance with multiple dev-servers, e.g.:
```js
const compiler = webpack(config);
const server1 = new Server(compiler, options1);
server1.close(() => {
  const server2 = new Server(compiler, options2);
});
```
Should this be supported or disallowed?

Reply:
supported, but with only `apply` it should work fine

Reply:
But if options change, entires should be changed, right?

Reply:
Yes, right, options should be updated